### DISCUSSION TITLE
Trigger error() in +page.server.ts

### DISCUSSION THREAD

**Discussion by jonaskohl** [sveltejs/kit#11701] [Q&A]
I have a page with the paths `[slug]/+page.svelte` and `[slug]/+page.server.ts`. I also have a `+error.svelte` error page. Inside the `+page.server.ts` I'm fetching the page contents from a CMS. If the slug does not exist inside the CMS, I want to trigger a 404 page with `error(404, "Not found")`. But for some reason, the error call gets ignored completely.

Now, I've read that `error()` does not seem to be supported in `*.server.ts` files (which is only mentioned as a little disclaimer here: https://kit.svelte.dev/docs/load#errors and not at all here: https://kit.svelte.dev/docs/errors). What would be the best approach here?

(Note: I don't want to import `tryFetchPageFromCMS` into client code as I don't want to compile and leak CMS authentication details to the client.)

My `+page.server.ts` for reference:

```typescript
import { tryFetchPageFromCMS } from "$lib/server/cms";
import { error } from "@sveltejs/kit";

export async function load({ params }) {
    const page = await tryFetchPageFromCMS(params.slug);
    if (!page) error(404, "Not found");
    return {
        page
    };
}
```

**Comment by david-plugge** [comment]
I'm sorry, but where did you take that information from? I can't find it anywhere.
`error(...)` works in every type of load function!

**Comment by jonaskohl** [comment]
This: https://kit.svelte.dev/docs/routing#error page mentions it briefly:
> +error.svelte is not used when an error occurs inside [handle](https://kit.svelte.dev/docs/hooks#server-hooks-handle) or a [+server.js](https://kit.svelte.dev/docs/routing#server) request handler.
